### PR TITLE
fix: subdomains should be [] if the host is an ip

### LINKED ISSUE #775: this.subdomains is error parsed
when host is 127.0.0.1 this.subdomains=> [127,0,0]

### PR DESCRIPTION
**Pull request by welladamm**
Fixes #775

### CODE REVIEW

File: lib/request.js [hunk 1]
```diff
@@ -417,7 +417,13 @@ module.exports = {
 
   get subdomains() {
     var offset = this.app.subdomainOffset;
-    return (this.host || '')
+    var ip = /^(\d{1,3}\.){3}\d{1,3}$/gim;
```
Comment (PlasmaPower) [inline lib/request.js]: This doesn't accommodate IPv6. Also, I don't like the use of a regex here, is there a better way to do this?

Comment (PlasmaPower) [inline lib/request.js]: If you do keep the regex, remove the modifiers, they are all irrelevant.

**Comment by PlasmaPower** [comment]
:+1: on premise, but I don't like the current implementation.

File: lib/request.js (hunk 1 above)
Comment (welladamm) [inline lib/request.js]: I don't think IPv6 is a big concern for the moment. However I did think about leaving a TODO in the comment, but, hey, using IPv4 address as the host is ugly already, who'd bother with IPv6 anyway.

Comment (welladamm) [inline lib/request.js]: AFAIK, this regex might just be a pragmatic solution, but I'm open to better suggestions.

Comment (welladamm) [inline lib/request.js]: You are right about the modifers. I'll fix this when I'm on my laptop.

Comment (tj) [inline lib/request.js]: Looks like there's an is-ip module in npm, might as well delegate!

Comment (dougwilson) [inline lib/request.js]: Isn't this part of the code Node.js `net` library? https://nodejs.org/dist/latest-v4.x/docs/api/net.html#net_net_isip_input

Comment (PlasmaPower) [inline lib/request.js]: Looks like exactly what we need.

**Comment by welladamm** [comment]
Shoot, totally forgot there's a native implementation. My bad.

**Comment by dead-horse** [comment]
LGTM

**Comment by dead-horse** [comment]
@welladamm make a PR to v2.x ?

**Comment by welladamm** [comment]
@dead-horse Will do.

**Pull request merged**